lib/ews/connection.rb

```diff
@@ -216,14 +216,15 @@ def post_async(xmldoc, opts: {})
 
   def check_response(resp, include_http_headers: false, request_body: nil, options: {})
     @log.debug("Got HTTP response with headers: #{resp.headers}")
+    @log.debug("Got the following 'Set-Cookie' Values #{resp.header['Set-Cookie']}")
     @log.debug("Got HTTP response with body: #{resp.body}") if resp.body
 
     log msg: "Received a **#{resp.status}** response for request of type: **#{options[:request_type]}** with uniq id: **#{options[:uniq_id]}** with body: #{resp.body}"
 
     case resp.status
     when 200
       if include_http_headers
-        return resp.body, resp.headers
+        return resp.body, resp.headers, resp.header['Set-Cookie']
```

Comment:
there are 4 possible cookie paths `["Set-Cookie", "set-cookie", "Cookie", "cookie"]` do we want to parse all of them rather than just this one?

Reply:
From my testing it has only ever been "Set-Cookie" these are the ones that are flattened

Reply:
```
   # Returns Hash of header. key and value are both String. Each key has a
    # single value so you can't extract exact value when a message has multiple
    # headers like 'Set-Cookie'. Use header['Set-Cookie'] for that purpose.
    # (It returns an Array always)
    def headers
      Hash[*http_header.all.flatten]
    end
```

Reply:
@iambanklee do you have a link to something that states this? I wanna make sure we get this right.

Reply:
FYI @iambanklee @richkettle https://docs.microsoft.com/en-us/exchange/client-developer/exchange-web-services/how-to-maintain-affinity-between-group-of-subscriptions-and-mailbox-server the sample in here seems to have it as "Set-Cookie"

<img width="867" alt="Screenshot 2019-11-12 at 15 16 52" src="https://user-images.githubusercontent.com/2750491/68683882-77e8d100-055f-11ea-9462-5048a4f4a4b4.png">

Reply:
@richkettle no i don't have but these were found when we implementing it